### PR TITLE
feat(ui): Implement Toast new design #27369

### LINKED ISSUE #27369: Design System: Implement Toast Component
### Parent Issue

_No response_

### Task

Apply styles as designed in [figma](https://www.figma.com/file/7CzbMcGSbTIJjerWtcapAm/Design-System-2023?node-id=1118:11576&mode=dev) and update Storybook Component for Toast

Screenshot 2024-01-18 at 2 19 29 PM

### Proposed Objective

User Experience

### Proposed Priority

Please Select

### Acceptance Criteria

* Design should be matching the figma file
* Storybook should be updated
* Toast should work as expected after the redesign

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

- [Figma Design](https://www.figma.com/file/7CzbMcGSbTIJjerWtcapAm/Design-System-2023?node-id=1118:11576&mode=dev)
- [PrimeNG Toast](https://www.primefaces.org/primeng-v15-lts/toast#api.props)

### Assumptions & Initiation Needs

* Toast component already has a global styles file
* Toast component already has a Storybook component
* Toast component is a PrimeNG component

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by zJaaal**
### Proposed Changes
* Update toast styles
* Update Storybook component

## Screenshots

https://github.com/dotCMS/core/assets/63567962/6e0478cb-94e4-48c8-a368-bf0c618a3308

https://github.com/dotCMS/core/assets/63567962/a309c9e9-56f4-4788-b87b-c836c7f3b6f9

### CODE REVIEW

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    height: fit-content;
+    border-radius: $border-radius-md;
+
+    .p-toast-message-content {
+        padding: $spacing-3;
+        padding-bottom: $spacing-4;
+        display: flex;
+        align-items: flex-start;
+        gap: $spacing-1;
+        border-radius: $border-radius-md;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:19]:
```suggestion
        border-radius: $border-radius-md;
        border-width: 8px;
        border-style: solid;
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+            &:focus {
+                background-color: transparent;
+                @include field-focus;
+            }
+        }
+    }
+
+    &.p-toast-message-success .p-toast-message-content {
+        border-bottom: $toast-bottom-border $color-alert-green;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:93]:
```suggestion
        border-color: $color-alert-green;
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+        .p-toast-message-icon {
+            color: $color-alert-green;
+        }
+    }
+
+    &.p-toast-message-error .p-toast-message-content {
+        border-bottom: $toast-bottom-border $color-alert-red;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:101]:
```suggestion
        border-color: $color-alert-red;
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+        .p-toast-message-icon {
+            color: $color-alert-red;
+        }
+    }
+
+    &.p-toast-message-info .p-toast-message-content {
+        border-bottom: $toast-bottom-border $color-palette-primary-500;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:109]:
```suggestion
        border-color: $color-palette-primary-500;
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+        .p-toast-message-icon {
+            color: $color-palette-primary-500;
+        }
+    }
+    &.p-toast-message-warn .p-toast-message-content {
+        border-bottom: $toast-bottom-border $color-alert-yellow;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:116]:
```suggestion
        border-color: $color-alert-yellow;
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            font-size: $font-size-md;
+
+            .p-toast-summary,
+            .p-toast-detail {
+                align-self: stretch;
+                line-height: 140%;
+            }
+
+            .p-toast-summary {
+                height: 1.375rem;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:52]: Why?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    .p-toast-message-content {
+        padding: $spacing-3;
+        padding-bottom: $spacing-4;
+        display: flex;
+        align-items: flex-start;
+        gap: $spacing-1;
+        border-radius: $border-radius-md;
+
+        .p-toast-message-icon {
+            height: 1.375rem;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:22]: Is not squared?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/stories/primeng/messages/Toast.component.ts [hunk 8]
```diff
@@ -1,120 +1,38 @@
...
-        this.messageService.add({
-            severity: 'error',
-            detail: 'Something went wrong, please try again.'
-        });
-    }
+    @Input() severity: string = 'success';
+    @Input() summary: string = 'Success Message';
+    @Input() detail: string = 'The action "Publish" was executed succesfully';
+    @Input() position: string = 'top-right';
+    @Input() life: number = 2000;
```
Comment (KevinDavilaDotCMS) [inline core-web/apps/dotcms-ui/src/stories/primeng/messages/Toast.component.ts:25]: These inputs can be only one input object? Or is toast being used in many places and would complicate the refactor?

Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/stories/primeng/messages/Toast.component.ts:25]: this is just a component for storybook, we need the inputs separated to pass them using storybook functionality, not worth a refactor

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss (hunk 7 above)
Comment (zJaaal) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:22]: it is, but i need it to be that height to center it with the text

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/stories/primeng/messages/Toast.component.ts
[the remark this answers is not in the document]
Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/stories/primeng/messages/Toast.component.ts:1]: Done!

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss (hunk 7 above)
Comment (fmontes) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:24]: But if the text is multiples lines, we should just add a top padding prob?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss (hunk 7 above)
Comment (zJaaal) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:24]: Not sure how to do it. I was testing multiline and it looks like every line has a height of 22.4px

Comment (zJaaal) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/messages/_toast.scss:24]: Not sure how to do it. I was testing multiline and it looks like every line has a height of 22.4px

**Review by fmontes — APPROVED** [review]

**Pull request merged**